tests/suites/options.js

```diff
@@ -641,3 +641,54 @@ test('Multidate Separator', function(){
     target.click();
     equal(input.val(), '2012-03-05 2012-03-04 2012-03-12');
 });
+
+
+test("Picker is shown on input focus when showOnFocus is not defined", function () {
+
+    var input = $('<input />')
+            .appendTo('#qunit-fixture')
+            .val('2014-01-01')
+            .datepicker({
+            }),
+        dp = input.data('datepicker'),
+        picker = dp.picker;
+
+    input.focus();
+
+    ok(picker.is(":visible"), "Datepicker is hidden");
+
+});
+
+test("Picker is shown on input focus when showOnFocus is true", function () {
+
+    var input = $('<input />')
+            .appendTo('#qunit-fixture')
+            .val('2014-01-01')
+            .datepicker({
+                showOnFocus: true
+            }),
+        dp = input.data('datepicker'),
+        picker = dp.picker;
+
+    input.focus();
+
+    ok(picker.is(":visible"), "Datepicker is hidden");
+
+});
+
+test("Picker is hidden on input focus when showOnFocus is false", function () {
+
+    var input = $('<input />')
+            .appendTo('#qunit-fixture')
+            .val('2014-01-01')
+            .datepicker({
+                showOnFocus: false
+            }),
+        dp = input.data('datepicker'),
+        picker = dp.picker;
+
+    input.focus();
+
+    ok(picker.is(":hidden"), "Datepicker is visible");
```

Comment:
Hi, i believe the message is the message displayed if the test fails - i.e.
it should indicate the incorrect state of the value you are asserting.
However, I am not very familiar with this testing framework, and based my
tests on the previous tests in the suite, so please correct me if I am
wrong :)

On Sat Jan 03 2015 at 3:33:16 PM Jeroen Thora <notifications@github.com>
wrote:

> In tests/suites/options.js
> <https://github.com/eternicode/bootstrap-datepicker/pull/802#discussion-diff-22431799>
> :
>
> > +});
> > +
> > +test("Picker is hidden on input focus when showOnFocus is false", function () {
> > +
> > +    var input = $('<input />')
> > +            .appendTo('#qunit-fixture')
> > +            .val('2014-01-01')
> > +            .datepicker({
> > +                showOnFocus: false
> > +            }),
> > +        dp = input.data('datepicker'),
> > +        picker = dp.picker;
> > +
> > +    input.focus();
> > +
> > +    ok(picker.is(":hidden"), "Datepicker is visible");
>
> Shouldn't this message state "Datepicker is hidden"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eternicode/bootstrap-datepicker/pull/802/files#r22431799>
> .
>